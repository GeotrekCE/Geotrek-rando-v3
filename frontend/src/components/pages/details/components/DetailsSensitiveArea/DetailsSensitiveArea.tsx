import parse from 'html-react-parser';
import { SensitiveArea } from 'modules/sensitiveArea/interface';
import { FormattedDate, FormattedMessage } from 'react-intl';
import styled from 'styled-components';
import { borderRadius, colorPalette, getSpacing } from 'stylesheet';
import { StyledLink } from '../../../../../components/Link';
import { HtmlText } from '../../utils';

type DetailsSensitiveAreaProps = Omit<SensitiveArea, 'geometry'> & { className?: string };

export const DetailsSensitiveArea: React.FC<DetailsSensitiveAreaProps> = ({
  name,
  className,
  contact,
  infoUrl,
  description,
  period,
  practices,
  color,
}) => {
  const hasPeriodAtLeastOneMonthValid = period?.some(monthlyValidity => monthlyValidity === true);
  return (
    <div id="details_sensitiveArea" className={className}>
      <div className="flex items-center space-x-2" id="details_sensitiveAreaTitle">
        <ColorLegendIcon color={color} />
        {name !== null && <span className="font-bold text-H4 space-y-2">{name}</span>}
      </div>
      {description !== null && (
        <SensitiveAreaSection>
          <HtmlText>{parse(description)}</HtmlText>
        </SensitiveAreaSection>
      )}
      {practices.length > 0 && (
        <SensitiveAreaSection labelId="practices">
          <div>{practices.map(practice => practice.name).join(', ')}</div>
        </SensitiveAreaSection>
      )}
      {period !== null && hasPeriodAtLeastOneMonthValid && (
        <SensitiveAreaSection labelId="period">
          <div>
            {period.map((monthlyValidity, i) =>
              monthlyValidity ? <StyledMonth monthNumber={i} key={i} /> : undefined,
            )}
          </div>
        </SensitiveAreaSection>
      )}
      {contact !== null && (
        <SensitiveAreaSection labelId="contact">
          <HtmlText>{parse(contact)}</HtmlText>
        </SensitiveAreaSection>
      )}
      {infoUrl !== null && (
        <SensitiveAreaSection>
          <StyledLink href={infoUrl}>
            <FormattedMessage id={'details.knowMore'} />
          </StyledLink>
        </SensitiveAreaSection>
      )}
    </div>
  );
};

const ColorLegendIcon = styled.div<{ color: string }>`
  background-color: ${({ color }) => color};
  border-radius: ${borderRadius.medium};
  border: 2px solid ${colorPalette.greyDarkColored};
  height: ${getSpacing(6)};
  width: ${getSpacing(6)};
`;
interface SensitiveAreaSectionProps {
  labelId?: string;
  children: React.ReactNode;
}

const SensitiveAreaSection: React.FC<SensitiveAreaSectionProps> = ({ labelId, children }) => {
  return (
    <div className="mt-1 desktop:mt-2">
      {labelId !== undefined && (
        <div className="font-bold">
          <FormattedMessage id={`details.sensitiveAreas.${labelId}`} />
        </div>
      )}
      {children}
    </div>
  );
};

interface FormattedMonthProps {
  monthNumber: number;
}

const StyledMonth: React.FC<FormattedMonthProps> = ({ monthNumber }) => {
  return (
    <span className="mr-2">
      <FormattedDate value={new Date(1990, monthNumber, 1)} month="short" />
    </span>
  );
};
