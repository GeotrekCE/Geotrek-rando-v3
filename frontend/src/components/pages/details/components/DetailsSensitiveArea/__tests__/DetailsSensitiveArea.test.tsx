import { render } from 'services/testing/reactTestingLibraryWrapper';
import { DetailsSensitiveArea } from '../DetailsSensitiveArea';
describe('DetailsSensitiveArea', () => {
  it('should display a DetailsSensitiveArea', () => {
    const component = render(
      <DetailsSensitiveArea
        name="Zone sensible de test"
        description="<p>Cette zone est tr√®s sensible</p>"
        contact={'<h2>Une personne a contacter</h2>'}
        infoUrl={'http://ensavoirplusici.com'}
        period={[true, true, false, false, false, false, false, false, false, false, false, false]}
      />,
    );

    expect(component).toMatchSnapshot();
  });
});
