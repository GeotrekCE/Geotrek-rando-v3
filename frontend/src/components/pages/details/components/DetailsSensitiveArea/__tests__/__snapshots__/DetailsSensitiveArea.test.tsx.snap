// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsSensitiveArea should display a DetailsSensitiveArea 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="details_sensitiveArea"
      >
        <div
          class="flex items-center space-x-2"
          id="details_sensitiveAreaTitle"
        >
          <div
            class="sc-dlfnuX dDiEiS"
            color="#AAAAA"
          />
          <span
            class="font-bold text-H4 space-y-2"
          >
            Zone sensible de test
          </span>
        </div>
        <div
          class="mt-1 desktop:mt-2"
        >
          <div
            class="sc-gsTEea dPWxIN"
          >
            <p>
              Cette zone est très sensible
            </p>
          </div>
        </div>
        <div
          class="mt-1 desktop:mt-2"
        >
          <div
            class="font-bold"
          >
            Domaines d'activités concernés :
          </div>
          <div>
            Terrestre, Aquatique
          </div>
        </div>
        <div
          class="mt-1 desktop:mt-2"
        >
          <div
            class="font-bold"
          >
            Périodes de sensibilité :
          </div>
          <div>
            <span
              class="mr-2"
            >
              janv.
            </span>
            <span
              class="mr-2"
            >
              févr.
            </span>
          </div>
        </div>
        <div
          class="mt-1 desktop:mt-2"
        >
          <div
            class="font-bold"
          >
            Contact :
          </div>
          <div
            class="sc-gsTEea dPWxIN"
          >
            <h2>
              Une personne a contacter
            </h2>
          </div>
        </div>
        <div
          class="mt-1 desktop:mt-2"
        >
          <a
            class="sc-bdfBQB btWAAI"
            href="http://ensavoirplusici.com"
          >
            En savoir plus
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="details_sensitiveArea"
    >
      <div
        class="flex items-center space-x-2"
        id="details_sensitiveAreaTitle"
      >
        <div
          class="sc-dlfnuX dDiEiS"
          color="#AAAAA"
        />
        <span
          class="font-bold text-H4 space-y-2"
        >
          Zone sensible de test
        </span>
      </div>
      <div
        class="mt-1 desktop:mt-2"
      >
        <div
          class="sc-gsTEea dPWxIN"
        >
          <p>
            Cette zone est très sensible
          </p>
        </div>
      </div>
      <div
        class="mt-1 desktop:mt-2"
      >
        <div
          class="font-bold"
        >
          Domaines d'activités concernés :
        </div>
        <div>
          Terrestre, Aquatique
        </div>
      </div>
      <div
        class="mt-1 desktop:mt-2"
      >
        <div
          class="font-bold"
        >
          Périodes de sensibilité :
        </div>
        <div>
          <span
            class="mr-2"
          >
            janv.
          </span>
          <span
            class="mr-2"
          >
            févr.
          </span>
        </div>
      </div>
      <div
        class="mt-1 desktop:mt-2"
      >
        <div
          class="font-bold"
        >
          Contact :
        </div>
        <div
          class="sc-gsTEea dPWxIN"
        >
          <h2>
            Une personne a contacter
          </h2>
        </div>
      </div>
      <div
        class="mt-1 desktop:mt-2"
      >
        <a
          class="sc-bdfBQB btWAAI"
          href="http://ensavoirplusici.com"
        >
          En savoir plus
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
